tests/test_ts_inference.py

```diff
@@ -0,0 +1,50 @@
+import os
+from pathlib import Path
+from packaging import version
+
+import torch
+import numpy as np
+from PIL import Image
+
+
+def load_image(imfile):
+    """Utility function to load image files as tensors"""
+    img = Image.open(imfile)
+    img = np.array(img).astype(np.uint8)
+    img = torch.from_numpy(img).permute(2, 0, 1).float()
+    img = img.to("cuda")
+
+    assert img.ndim == 3, f"Expected 3D tensor, got shape {img.shape}"
+    return img
+
+
+def test_monster_mix_inference():
+    # Turn off deprecated nvfuser if Torch version is old.
+    # Displays multiple warnings otherwise.
+    if version.parse(torch.__version__) < version.parse("2.2.0"):    
+        torch._C._jit_set_nvfuser_enabled(False)
+
+    with torch.no_grad():
+        model_path = Path('output') / 'monster-mix-script.pt'
+        assert os.path.exists(model_path), f"Model file not found at {model_path}"
+
+        model = torch.jit.load(model_path, map_location="cuda")
+        model.eval()
+
+        image_left_path = Path('input') / 'left' / 'sample_l.png'
+        image_right_path = Path('input') / 'right' / 'sample_r.png'
+
+        assert os.path.exists(image_left_path), f"Left image not found at {image_left_path}"
+        assert os.path.exists(image_right_path), f"Right image not found at {image_right_path}"
+
+        image_left = load_image(image_left_path)
+        image_right = load_image(image_right_path)
+
+        # Note: Use two warmup runs in production.
```

Comment:
Why?

Reply:
The second run seems to take the longest. The second run performs optimizations based on internal profiling for the first run by the executor.

It is the default number of configured warm up runs in our rust code too; in [torch-detection-server](https://github.com/fizyr-private/fizyr-torch-detection-server/blob/12070b2e779cc1e9cb043f9b6d2eba2445850da7/utils/src/config.rs#L119), and in [stereo-matching](https://github.com/fizyr-private/fizyr-stereo-matching/blob/956f51fcfcadacd404c01abcc94f7179d5bbc533/src/config.rs#L7) for example.

Reply:
Yes, the second prediction run took the longest time to execute. On a system with an RTX 3090, for a 1280x960 stereo pair, the first run took about 2 seconds, the second run took about 7 seconds, and from the third run onward it took 1.7 seconds.